File: codeup/100000571/E.cpp
#include <iostream>
using namespace std;

void my_input(int *list) {
  for (int i = 0; i < 10; i++) scanf("%d", &list[i]);
  return;
}

void deal_list(int *list) {
  int max_i, min_i, temp;
  max_i = min_i = 0;
  for (int i = 0; i < 10; i++) {
    if (list[max_i] < list[i]) max_i = i;
    if (list[min_i] > list[i]) min_i = i;
  }
  temp = list[min_i], list[min_i] = list[0], list[0] = temp;
  temp = list[max_i], list[max_i] = list[9], list[9] = temp;
  return;
}

void my_output(int *list) {
  for (int i = 0; i < 9; i++) printf("%d ", list[i]);
  printf("%d\n", list[9]);
  return;
}

int main() {
  int list[10];
  my_input(list);
  deal_list(list);
  my_output(list);
  return 0;
}

File: codeup/100000571/E.py
#!/usr/bin/python3
def my_input():
    in_list = list(map(int, input().split()))
    return in_list


def deal_list(in_list):
    list_len = len(in_list)
    max_i = min_i = 0
    for i in range(1, list_len):
        if in_list[max_i] < in_list[i]:
            max_i = i
        if in_list[min_i] > in_list[i]:
            min_i = i
    in_list[min_i], in_list[0] = in_list[0], in_list[min_i]
    in_list[max_i], in_list[list_len-1] = in_list[list_len-1], in_list[max_i]


def my_output(in_list):
    list_len = len(in_list)-1
    for cnt in range(list_len):
        print(in_list[cnt], end=' ')
    print(in_list[list_len])


in_list = my_input()
deal_list(in_list)
my_output(in_list)
